File: 1029-two-city-scheduling/1029-two-city-scheduling.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        hmap = [[(a-b),[a,b]] for a, b in costs]
        hmap = sorted(hmap, key= lambda x: x[0], reverse=True)
        print(hmap)
        ans = 0
        n = 1
    
        for val in hmap:
            if n > len(costs)//2:
                ans += val[1][0]
                print(val[1][0])
            else:
                ans += val[1][1]
                print(val[1][1])
            n += 1
        return ans
        

File: 1029-two-city-scheduling/1029-two-city-scheduling.java
class Solution {
    public int twoCitySchedCost(int[][] costs) {
        Arrays.sort(costs, new ProfitComparator());
        int ans = 0;
        for(int i=0; i<costs.length; i++) {
            if(i < costs.length/2) {
                ans += costs[i][0];
            }
            else {
                ans += costs[i][1];
            }
        }
        return ans;
    }
    
    public class ProfitComparator implements Comparator<int[]> {
    @Override
    public int compare(int[] one, int[] two) {
        int profit1 = one[1]-one[0];
        int profit2 = two[1]-two[0];
        
        return profit2 - profit1;
    }
}
}

